src/python/grapl-plugin-sdk/grapl_plugin_sdk/analyzer/service_impl.py

```diff
@@ -60,6 +61,21 @@ class PropertyUpdate(Protocol):
 )
 
 
+class RunAnalyzerRequestMetadata:
+    trace_id: str | None
+
+    def __init__(self, grpc_metadata: grpc_aio.Metadata | None) -> None:
+        grpc_metadata = grpc_metadata or {}
+        self.trace_id = grpc_metadata.get("x-trace-id")
+
+    def to_grpc_metadata(self) -> GrpcOutboundMetadata:
+        metadata = [
+            ("x-trace-id", self.trace_id),
+        ]
+        metadata_no_nones = [(k, v) for (k, v) in metadata if v is not None]
```

Comment:
Is more going to be added to `metadata` at any point? Otherwise it seems like all this could be simplified to
```python
return (("x-trace-id", self.trace_id)) if self.trace_id else ()
```

Reply:
This question probably just disappears if metadata is modeled as a dict to begin with:
```python
metadata = {}
if self.trace_id:
    metadata["x-trace-id"] = self.trace_id
# add more as needed
return metadata
```